### PR TITLE
feature/more documentation fixes

### PR DESCRIPTION
**Pull request by anders-wind**
- Updated readme
- Added size, iterate and index skills
- Fix hacking document

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -119,21 +119,23 @@ struct Euro : twig::stronk<Euro, double, twig::unit>
...
-    auto watt_hours_per_euro = 1. / euros_per_watt_hour;  // `(Watt * Hours) / Euro`
-    WattHours watt_hours_affordable_for_500_euros = watt_hours_per_euro * Euro {500.};
+    auto mega_watt_hours_per_euro = 1. / euros_per_mega_watt_hour;  // `(Watt * Hours) / Euro`
+    WattHours mega_watt_hours_affordable_for_500_euros = mega_watt_hours_per_euro * Euro {500.};
 }
 ```
 
 Units are a great way of using the type system to validate your code.
 
+Credit to https://www.foonathan.net/2016/10/strong-typedefs/ for a great amount of inspiration
```
Comment (engesgaard) [inline README.md:137]: You could also consider linking his GitHub repo: https://github.com/foonathan/type_safe. You could also put this under why, and write how stronk supplements his library

File: include/stronk/stronk.h [hunk 2]
```diff
@@ -374,6 +374,82 @@ struct can_hash
...
+template<typename StronkT>
+struct can_size
+{
+    [[nodiscard]] constexpr auto size() const noexcept -> std::size_t
+    {
+        return static_cast<const StronkT&>(*this).template unwrap<StronkT>().size();
+    }
+    [[nodiscard]] constexpr auto empty() const noexcept -> bool
+    {
+        return this->size() == 0ULL;
```
Comment (engesgaard) [inline include/stronk/stronk.h:386]: How will this behave on a 32-bit system? :-)

File: tests/src/stronk_tests.cpp [hunk 3]
```diff
@@ -424,4 +425,112 @@ TEST(can_hash, can_hash_overloads_std_hash)  // NOLINT
...
+
+struct a_can_iterate_vector_type : stronk<a_can_iterate_vector_type, std::vector<int>, can_iterate>
+{
+    using stronk::stronk;
+};
+
+TEST(can_const_iterate, can_const_iterate_works_for_vectors)
+{
+    for (size_t i = 0; i < 16; i++) {
+        const auto vector = [&i]()
```
Comment (engesgaard) [inline tests/src/stronk_tests.cpp:464]: What determines that `vector` is of type `a_can_iterate_vector_type`?

File: include/stronk/stronk.h (hunk 2 above)
Comment (anders-wind) [reply include/stronk/stronk.h:386]: I think it will work?

Comment (anders-wind) [reply include/stronk/stronk.h:386]: https://stackoverflow.com/a/65934742 <- in 23 we get size_t literal.

File: tests/src/stronk_tests.cpp (hunk 3 above)
Comment (anders-wind) [reply tests/src/stronk_tests.cpp:464]: omg im noob

File: README.md [hunk 4]
```diff
@@ -119,21 +119,23 @@ struct Euro : twig::stronk<Euro, double, twig::unit>
...
-    auto watt_hours_per_euro = 1. / euros_per_watt_hour;  // `(Watt * Hours) / Euro`
-    WattHours watt_hours_affordable_for_500_euros = watt_hours_per_euro * Euro {500.};
+    auto mega_watt_hours_per_euro = 1. / euros_per_mega_watt_hour;  // `(Watt * Hours) / Euro`
+    WattHours mega_watt_hours_affordable_for_500_euros = mega_watt_hours_per_euro * Euro {500.};
 }
 ```
 
 Units are a great way of using the type system to validate your code.
 
+Credit to [Jonathan Müller](https://github.com/foonathan)'s [blogpost](https://www.foonathan.net/2016/10/strong-typedefs/) and [Jonathan Boccara](https://github.com/joboccara)'s [blogpost](https://www.fluentcpp.com/2016/12/08/strong-types-for-strong-interfaces/) - both of which has been great sources of inspiration.
```
Comment (engesgaard) [inline README.md:137]: I think it is "both of which have" 🤓

**Review by engesgaard — APPROVED** [review]

**Review by engesgaard — APPROVED** [review]

**Pull request merged**